src/main/java/org/chameleoncloud/api/admin/UserGroupRolesResource.java

```diff
@@ -0,0 +1,320 @@
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.chameleoncloud.representations.UserGroupRoles;
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.keycloak.authorization.model.Policy;
+import org.keycloak.authorization.model.ResourceServer;
+import org.keycloak.authorization.model.Scope;
+import org.keycloak.authorization.store.PolicyStore;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.authorization.GroupPolicyRepresentation;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionManagement;
+import org.keycloak.services.resources.admin.permissions.AdminPermissions;
+import org.keycloak.services.resources.admin.permissions.GroupPermissionManagement;
+
+public class UserGroupRolesResource {
+
+    private AdminPermissionEvaluator auth;
+    private KeycloakSession session;
+    private RealmModel realm;
+    private AdminPermissionManagement adminPermissionManagement;
+    private GroupPermissionManagement groupPermission;
+    private PolicyStore policyStore;
+
+    public UserGroupRolesResource(KeycloakSession session, AdminPermissionEvaluator auth) {
+	this.auth = auth;
+	this.session = session;
+	this.realm = session.getContext().getRealm();
+	this.adminPermissionManagement = AdminPermissions.management(session, realm);
+	this.groupPermission = adminPermissionManagement.groups();
+	this.policyStore = adminPermissionManagement.authz().getStoreFactory().getPolicyStore();
+    }
+
+    private Set<String> getGroupIdsFromPolicyConfig(Map<String, String> policyConfig) {
+
+	// policy config returns a type of Map<String, String>
+	// e.g.
+	// {"groups":"[{\"id\":\"b770dfde-038b-493e-bc7f-a51a5ce86741\",\"extendChildren\":true}]"}
+	// this function parses the value of "groups" and extract the group ids into a
+	// hashset
+
+	Set<String> result = new HashSet<String>();
+	if (policyConfig.containsKey("groups")) {
+	    String regex = "[a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8}";
+	    Matcher m = Pattern.compile("(?=(" + regex + "))").matcher(policyConfig.get("groups"));
+	    while (m.find()) {
+		result.add(m.group(1));
+	    }
+	}
+	return result;
+    }
+
+    private SortedMap<Integer, String> getPolicies(Policy permission, Set<String> groupIds) {
+
+	// get the names of the group type policies from a permission
+	// and check if a user applies to the policies by comparing the group ids
+
+	SortedMap<Integer, String> result = new TreeMap<Integer, String>();
+	Map<String, Set<String>> policyGroups = new HashMap<String, Set<String>>();
+	permission.getAssociatedPolicies().stream().filter(policy -> "group".equals(policy.getType()))
```

Comment:
nit: `"group".equals(policy.getType())` looks so odd to me I guess b/c of the string literal. maybe it would look less weird like this?

```
policy.getType().equals(GROUP_POLICY_TYPE)
```
where GROUP_POLICY_TYPE is a string constant?